src/api/process.c

```diff
@@ -633,17 +633,24 @@ static int g_read(lua_State* L, int stream, unsigned long read_size) {
   #else
     luaL_Buffer b;
     luaL_buffinit(L, &b);
-    uint8_t* buffer = (uint8_t*)luaL_prepbuffsize(&b, READ_BUF_SIZE);
-    length = read(self->child_pipes[stream][0], buffer, read_size > READ_BUF_SIZE ? READ_BUF_SIZE : read_size);
-    if (length == 0 && !poll_process(self, WAIT_NONE))
-      return 0;
-    else if (length < 0 && (errno == EAGAIN || errno == EWOULDBLOCK))
-      length = 0;
-    if (length < 0) {
-      signal_process(self, SIGNAL_TERM);
-      return 0;
-    }
-    luaL_addsize(&b, length);
+    size_t offset = 0;
+    do {
+      uint8_t* buffer = (uint8_t*)luaL_prepbuffer(&b);
```

Comment:
Would it make sense to use `luaL_prepbuffsize` with size `min(read_size, LUAL_BUFFERSIZE)`?

Reply:
I think that's a 5.3 thing; so this was done with LUA_BUFFERSIZE rather than a custom buffer size to retain compatibility with jit.

Doesn't really change much, so I vote we just keep it to make jgm's life easier; it doesn't really change anything.

Reply:
It's fine for this, but we can't continue trying to keep compatibility with Lua versions we don't support, or we'll never use the modern features...

Reply:
Broadly agreed, but if there's no real reason to change it, may as well not bother.